rpms/build.sh

```diff
@@ -51,11 +58,13 @@ $linux rpmbuild --define "_topdir ${rpm_build_dir}" --define "_threads ${threads
         --define "_debugenabled ${debug}" --target ${architecture_target} \
         -ba ${rpm_build_dir}/SPECS/${package_name}.spec
 
-rpm_file="wazuh-${build_target}-${wazuh_version}-${package_release}.${architecture_target}.rpm"
-pkg_path="${rpm_build_dir}/RPMS/${architecture_target}"
-
 if [[ "${checksum}" == "yes" ]]; then
     cd ${pkg_path} && sha512sum ${rpm_file} > /var/local/checksum/${rpm_file}.sha512
+    cd ${src_path} && sha512sum ${src_file} > /var/local/checksum/${src_file}.sha512
+fi
+
+if [[ "${src}" == "yes" ]]; then
+    extract_path="${rpm_build_dir}"
 fi
 
-mv ${pkg_path}/${rpm_file} /var/local/wazuh
\ No newline at end of file
+find ${extract_path} -maxdepth 3 -type f -name "${file_name}*" -exec mv {} /var/local/wazuh \;
```

Comment:
This line extracts the src.rpm package even if the parameter is set or not. Check this out:
```shellsession
$ ls
build.sh  CentOS  generate_rpm_package.sh  README.md  SPECS
$ ./generate_rpm_package.sh -b v3.10.2 -t agent -j 8 -s $(pwd)
...
$ ls
build.sh  CentOS  generate_rpm_package.sh  README.md  SPECS  wazuh-agent-3.10.2-1.src.rpm
```
In addition, this doesn't extract the binary rpm package, just the sources.

Finally, I would remove this line completely and use something like this:
```
mv ${pkg_path}/${rpm_file} /var/local/wazuh
mv ${src_path}/${src_file} /var/local/wazuh
```

Reply:
Fixed in d20150a.